velox/docs/develop/types.rst

```diff
@@ -73,6 +73,26 @@ Examples:
 * Timestamp(-5000*24*60*60 - 1000, 123456) represents 1956-04-24 07:43:20.000123456
   (5000 days 1000 seconds before epoch plus 123456 nanoseconds).
 
+Floating point types (REAL, DOUBLE) have special values negative/positive infinity and
+not-a-number (NaN).
+For negative/positive infinity the following standard floating point semantics apply:
+
+* Positive infinity multiplied by any positive value returns positive infinity.
+* Negative infinity multiplied by any positive value returns negative infinity.
+* Positive infinity multiplied by any negative value returns negative infinity.
+* Negative infinity multiplied by any negative value returns positive infinity.
+* Positive/negative infinity multiplied by 0 returns NaN.
+* Positive/negative infinity is equal to itself.
+* Positive infinity and negative infinity are treated as normal values in join and group-by keys.
+* Positive infinity sorts lower than NaN and higher than any other values.
+* Negative infinity sorts lower than any other values.
+
+For NaN the semantics are different than the standard floating point semantics:
```

Comment:
Maybe state the special semantics for NaN before those for +/- infinity so they are atypical.